relay-server/src/actors/processor.rs

```diff
@@ -2284,61 +2284,33 @@ impl EnvelopeProcessorService {
     /// This execution of dynamic sampling is technically a "simulation" since we will use the result
     /// only for tagging errors and not for actually sampling incoming events.
     fn tag_error_with_sampling_decision(&self, state: &mut ProcessEnvelopeState) {
-        // In case there is no incoming event we can't tag anything, thus we early return.
         if state.event.is_empty() {
             return;
         }
 
-        // We want to run dynamic sampling only if we have a root project state and a dynamic
-        // sampling context.
-        //
-        // In reality the dynamic sampling logic supports optional root state and dsc but it will
-        // return keep. In our case having a keep in case of none root state and dsc will be
-        // a problem, since in reality we can't infer anything without trace metadata.
-        let sampling_result = if let (Some(root_project_state), Some(dsc)) = (
+        let sampled = utils::is_trace_fully_sampled(
+            self.config.processing_enabled(),
             state.sampling_project_state.as_deref(),
             state.envelope().dsc(),
-        ) {
-            utils::get_sampling_result(
-                self.config.processing_enabled(),
-                None,
-                Some(root_project_state),
-                Some(dsc),
-                None,
-            )
-        } else {
-            return;
-        };
+        );
 
-        let Some(event) = state.event.value_mut() else {
+        let (Some(event), Some(sampled)) = (state.event.value_mut(), sampled) else {
             return;
         };
 
-        // In case the sampling result is positive, we assume that all the transactions
-        // that have this DSC will be sampled and thus we mark the error as "having
-        // a full trace".
-        // In case we have no contexts object, we have to create it.
-        let contexts = event.contexts.get_or_insert_with(Contexts::new);
-
-        // We want to get the specific trace context, or we want to create it in case
-        // it is not there.
-        let context =
-            contexts.get_or_insert_with(TraceContext::default_key(), || Trace(Box::default()));
+        // We want to get the trace context, in which we will inject the `sampled` field.
+        let context = event
+            .contexts
+            .get_or_insert_with(Contexts::new)
+            .get_or_insert_with(TraceContext::default_key(), || Trace(Box::default()));
```

Comment:
Also here please check for the upstream diff, since there's a new `Contexts::insert_or_default()` and the import on enum variant `Trace` has been removed as per style guide.